### PR TITLE
new feature - pod stats

### PR DESCRIPTION
**Pull request by navidys**
pod_stats

Signed-off-by: Navid Yaghoobi <n.yaghoobi.s@gmail.com>

### CODE REVIEW

File: ui/pods/poddialogs/stats.go [hunk 1]
```diff
@@ -0,0 +1,482 @@
...
+	}
+	d.mu.Lock()
+	if index == 0 {
+		d.statQueryOpts.IDs = d.getAllPodIDs()
+	} else {
+		d.statQueryOpts.IDs = []string{
+			d.podDropDownOptions[index-1].ID,
+		}
+	}
+	defer d.mu.Unlock()
```
Comment (rhatdan) [inline ui/pods/poddialogs/stats.go:358]: Why isn't the defer right after the Lock?  Of if all you want to lock is the section between the section above, why use defer?

File: ui/pods/poddialogs/stats.go [hunk 2]
```diff
@@ -0,0 +1,482 @@
...
+	go d.query()
+}
+
+func (d *PodStatsDialog) setStatsQuerySortBy(name string, index int) {
+	if index == -1 {
+		return
+	}
+	d.mu.Lock()
+	d.statQueryOpts.SortBy = index
+	defer d.mu.Unlock()
```
Comment (rhatdan) [inline ui/pods/poddialogs/stats.go:368]: Why isn't the defer right after the Lock? Of if all you want to lock is the section between the section above, why use defer?

File: ui/pods/poddialogs/stats.go [hunk 3]
```diff
@@ -0,0 +1,482 @@
...
+				SetSelectable(false))
+	}
+	d.table.SetFixed(1, 1)
+	d.table.SetSelectable(true, false)
+}
+
+func (d *PodStatsDialog) updateData(statReport []ppods.StatReporter) {
+	d.mu.Lock()
+	d.statsResult = statReport
+	d.mu.Unlock()
```
Comment (rhatdan) [inline ui/pods/poddialogs/stats.go:402]: Why isn't the defer right after the Lock? Of if all you want to lock is the section between the section above, why use defer?

**Review by rhatdan — APPROVED** [review]
A couple of questions.

File: ui/pods/poddialogs/stats.go (hunk 2 above)
Comment (navidys) [reply ui/pods/poddialogs/stats.go:368]: Fixed.

File: ui/pods/poddialogs/stats.go (hunk 3 above)
Comment (navidys) [reply ui/pods/poddialogs/stats.go:402]: Here need section Lock, so I didn't use defer.

File: pdcs/pods/stats.go [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+		reporterItem.MemUsage = item.MemUsage
+		reporterItem.Mem = item.Mem
+		reporterItem.NetIO = item.NetIO
+		reporterItem.BlockIO = item.BlockIO
+		reporterItem.PIDS = item.PIDS
+		report = append(report, reporterItem)
+	}
+	sort.Slice(report, sortFunc(sortBy, report))
+	return report
+
```
[the remark this answers is not in the document]
Comment (navidys) [reply pdcs/pods/stats.go:66]: removed.

File: ui/pods/poddialogs/stats.go (hunk 1 above)
Comment (navidys) [reply ui/pods/poddialogs/stats.go:358]: Thanks for review, correct , it should be after Lock.

**Comment by rhatdan** [comment]
LGTM

**Pull request merged**